packages/odyssey-react-mui/src/labs/DataFilters.tsx

```diff
@@ -176,16 +176,50 @@ const DataFilters = ({
   }, [onChangeSearch, searchValue, searchDelayTime, hasSearchSubmitButton]);
 
   const handleInputChange = useCallback(
-    (filterId: string, value: DataFilterValue, submit: boolean = false) => {
-      setInputValues({ ...inputValues, [filterId]: value });
+    (
+      filterId: string,
+      value:
+        | string
+        | { label: string }
+        | null
+        | undefined
+        | (string | { label: string })[],
+      submit: boolean = false
+    ) => {
+      let extractedLabel: string | undefined = undefined;
+
+      if (Array.isArray(value)) {
+        // Extract the first valid string or { label: string } object from the array
+        const validValue = value.find(
+          (v) =>
+            typeof v === "string" || (typeof v === "object" && "label" in v)
+        );
+
+        if (validValue) {
+          extractedLabel =
+            typeof validValue === "string" ? validValue : validValue.label;
+        }
+      } else if (
```

Comment:
I don't have all the context here but, this seems like it could be just 
```js
} else if (value) {...
```
and then 
```js 
extractedLabel = typeof value === "string" ? value : value.label
```